core/modules/views/templates/views.theme.inc

```diff
@@ -676,126 +676,115 @@ function template_preprocess_views_view_table(&$variables) {
   $columns  = $options['columns'];
-
-  $rows = array();
-  $row_indexes = array();
-
-  if ($options['alignment'] == 'horizontal') {
-    $row = array();
-    $col_count = 0;
-    $row_count = 0;
-    $count = 0;
-    foreach ($variables['rows'] as $row_index => $item) {
-      $count++;
-      $row[] = $item;
-      $row_indexes[$row_count][$col_count] = $row_index;
-      $col_count++;
-      if ($count % $columns == 0) {
-        $rows[] = $row;
-        $row = array();
-        $col_count = 0;
-        $row_count++;
-      }
-    }
-    if ($row) {
-      // Fill up the last line only if it's configured, but this is default.
-      if (!empty($handler->options['fill_single_line']) && count($rows)) {
-        for ($i = 0; $i < ($columns - $col_count); $i++) {
-          $row[] = '';
-        }
-      }
-      $rows[] = $row;
-    }
-  }
-  else {
-    $num_rows = floor(count($variables['rows']) / $columns);
-    // The remainders are the 'odd' columns that are slightly longer.
-    $remainders = count($variables['rows']) % $columns;
-    $row = 0;
-    $col = 0;
-    foreach ($variables['rows'] as $count => $item) {
-      $rows[$row][$col] = $item;
-      $row_indexes[$row][$col] = $count;
-      $row++;
-
-      if (!$remainders && $row == $num_rows) {
-        $row = 0;
-        $col++;
-      }
-      elseif ($remainders && $row == $num_rows + 1) {
-        $row = 0;
-        $col++;
-        $remainders--;
-      }
-    }
-    for ($i = 0; $i < count($rows[0]); $i++) {
-      // This should be string so that's okay :)
-      if (!isset($rows[count($rows) - 1][$i])) {
-        $rows[count($rows) - 1][$i] = '';
-      }
-    }
+  $variables['classes_array'] = array();
+  if (!$options['default_classes']) {
+    $variables['classes_array'][] = 'views-grid';
+    $variables['classes_array'][] = 'views-grid-' . $options['alignment'];
+    $variables['classes_array'][] = 'views-columns-' . $columns;
   }
+  $variables['classes_array'] = array_merge($variables['classes_array'], explode(' ', $options['wrapper_classes']));
+  $variables['classes_array'] = array_filter($variables['classes_array']);
+  // Check if there are views results to process.
+  if (!empty($variables['rows'])) {
+    $col_id = 0;
+    $row_id = 0;
+    $keys = array_keys($variables['rows']);
+    // Determine style alignment.
+    if ($options['alignment'] == 'horizontal') {
+      $rows = array();
+      $variables['row_classes'] = array();
+      foreach ($variables['rows'] as $i => $row) {
+        $classes = explode(' ', $options['column_classes']);
+        if (!$options['default_classes']) {
+          $classes[] = 'views-column-' . ($col_id + 1);
+        }
+        $rows[$row_id][$col_id] = array(
+          'classes' => implode(' ', $classes),
+          'content' => $variables['rows'][current($keys)],
+        );
+        if ($options['automatic_width']) {
+          $rows[$row_id][$col_id]['col_style'] = array('style' => 'flex: ' . (100 / $columns) . '%;');
```

Comment:
I actually started with using core Bootstrap and also tried avoiding inline styles, but it seems unavoidable unless we limit the number of columns. FWIW D8 also uses inline styles.

We could apply a CSS style based on the number of columns, and define the width in CSS:
```
$rows[$row_id][$col_id]['col_style'] = 'col-number-' . $columns;
```
and CSS:
```
.col-number-5 { width: 20%;}
```
But we'd need a declaration for every possible column number, and would need to limit columns then.

If Bootstrap, we could add the Bootstrap classes to the colums easily. But for a 12-column grid like we have, once a user made 6 or more columns, we'd have to make all of them `col-md-1` because anything less wouldnt fit. i.e. 7 columns of `col-md-2` would be wider than one `col-md-12`.

Furthermore, if the user is insane and makes a grid with > 12 columns, Bootstrap doesn't account for this.

In short, inline styles or limit number of columns. Or JS.